app/controllers/katello/api/v2/repositories_controller.rb

```diff
@@ -544,5 +581,13 @@ def filter_by_content_view(query, content_view_id, environment_id, is_available_
       end
       query
     end
+
+    def generic_remote_options_hash(repo_params)
+      generic_remote_options = {}
+      RepositoryTypeManager.generic_remote_options.each do |option|
+        generic_remote_options[option.name] = repo_params[option.name]
```

Comment:
I think there might be an issue here when other generic types are introduced. `generic_remote_options` will have options from other repository types in it. I think you'll need to limit this to the repo type.